File: Swift/maximum-subarray.swift
class Solution {
    func maxSubArray(_ nums: [Int]) -> Int {
        var maxCurrent = nums[0]
        var maxGlobal = nums[0]
        
        for i in 1..<nums.count {
            maxCurrent = max(maxCurrent + nums[i], nums[i])
            maxGlobal = max(maxCurrent, maxGlobal)
        }
        
        return maxGlobal
    }
}

File: Python/maximum-subarray.py
class Solution:
    def maxSubArray1(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maxCureent = nums[0]
        maxGlobal = nums[0]

        for i in range(1, len(nums)):
            maxCureent = max(maxCureent + nums[i], nums[i])
            maxGlobal = max(maxCureent, maxGlobal)
        return maxGlobal
